### PR TITLE
Big *** user guide rework. 

### PR DESCRIPTION
**Pull request by JustinSGray**
Restructured the user-guide into chapters. Filled in the tutorials for the first two chapters. 

Also had to tweak a couple of tests.

### CODE REVIEW

File: openmdao/test_suite/components/paraboloid.py [hunk 1]
```diff
@@ -16,6 +16,9 @@ def setup(self):
 
         self.add_output('f_xy', val=0.0)
 
+        # Finite difference all partials.
+        self.approx_partials('*', '*', method='fd')
+
     def compute(self, inputs, outputs):
         """
```
Comment (Kenneth-T-Moore) [inline openmdao/test_suite/components/paraboloid.py]: Need to make your own test component rather than modifying this, since this one is used a bunch of places for specific purposes. Then you can get rid of the changes in paraboloid_mat_vec.

File: openmdao/test_suite/tests/test_feature_sellar.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 from openmdao.api import Problem
 from openmdao.devtools.testutil import assert_rel_error
-from openmdao.test_suite.components.sellar_feature import SellarNoDerivatives
+from openmdao.test_suite.components.sellar import SellarNoDerivatives
```
Comment (Kenneth-T-Moore) [inline openmdao/test_suite/tests/test_feature_sellar.py]: Why did you change the sellar_feature test to not test sellar_feature anymore?

File: openmdao/docs/index.rst [hunk 3]
```diff
@@ -2,21 +2,58 @@
...
+OpenMDAO is an open-source high-performance computing platform for systems analysis and multidisciplinary optimization, written in Python.
+It enables you to decompose your models, making them easier to build and maintain, while still solving them in a tightly coupled manner with efficient parallel numerical methods.
+
+The OpenMDAO project is primarily focused on supporting gradient based optimization with analytic derivatives to allow you to explore large design spaces with 100's or 1000's of design variables,
+but the framework also has a number of parallel computing features that can work with gradient free optimization, mixed integer nonlinear programming, and traditional design space exploration.
+
+User Guide
+***************************
+
+These are a collection of tutorial problems that teach you import concepts and techniques for using OpenMDAO.
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/index.rst]: import concepts -> important concepts

File: openmdao/docs/basic_guide/sellar.rst [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+    # Nonlinear Block Gauss Seidel is a gradient free solver
+    cycle.nonlinear_solver = NonlinearBlockGS()
+
+
+Our :code:`SellarMDA` Group, when instantiated, will have a three level hierarchy with itself as the top most level.
+
+.. figure:: images/sellar_tree.png
+   :align: center
+   :width: 80%
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/basic_guide/sellar.rst]: I think you might want to shrink this image down  abit.

File: openmdao/docs/basic_guide/promote_vs_connect.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+In the previous tutorial we built up a model of the Sellar problem using two disciplinary components and a few :code:`ExecComps`.
+In order to get OpenMDAO to pass the data between all the components,
+we linked everything up using promoted variables so that data passed from outputs to inputs with the same promoted name.
+
+Promoting variables is often a convenient way to establish the data passing links from outputs to inputs.
+However, you can also use calls to the :code:`connect` method in order to link outputs to inputs without having to promote anything.
+Here is how you would define the same Sellar model using:
+
+    #. Variable Promotion
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/basic_guide/promote_vs_connect.rst]: lower case p in promotion to make parallel

File: openmdao/docs/basic_guide/promote_vs_connect.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+.. embed-test::
+    openmdao.test_suite.test_examples.test_sellar_mda_promote_connect.TestSellarMDAPromoteConnect.test_sellar_mda_promote
+
+There are a few important details to note:
+
+    * the promoted name of an output has to be unique within that level of the hierarchy (i.e. you can't have two outputs with the same name)
+    * You are allowed to have multiple inputs promoted to the same name, but in order for a connection to be made there must also be an output with the same name. Otherwise, no connection is made.
+    * You can use glob-patterns to promote lots of variables without specifying them all, but try to limit your usage of :code:`promotes=['*']`.
+        Thought it may seem like a convinient way to do things, it can make it really hard for other people who are reading your code to understand what variables connect to where.
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/basic_guide/promote_vs_connect.rst]: Thought -> Though

File: openmdao/docs/basic_guide/promote_vs_connect.rst [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+Varoable Promotion
+********************
+
+.. embed-test::
+    openmdao.test_suite.test_examples.test_sellar_mda_promote_connect.TestSellarMDAPromoteConnect.test_sellar_mda_promote
+
+There are a few important details to note:
+
+    * the promoted name of an output has to be unique within that level of the hierarchy (i.e. you can't have two outputs with the same name)
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/basic_guide/promote_vs_connect.rst]: capitalize t in The

File: openmdao/docs/basic_guide/first_mdao.rst [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+.. embed-test::
+    openmdao.test_suite.test_examples.test_sellar_opt.TestSellarOpt.test_sellar_opt
+
+
+Controling the Solver Print Output
+************************************
+Notice the call to :code:`prob.set_solver_print()`,
+which sets the solver output to level 0.
+This is the semi-quiet setting where you only get notified if the solver failed to converge.
+There are lots of ways to :ref:`configure the solver print <solver-options>` output in your model to suite your needs.
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/basic_guide/first_mdao.rst]: suite your needs -> suit your needs

File: openmdao/docs/basic_guide/first_mdao.rst [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+Approximate the total derivatives with finite difference
+------------------------------------------------------------
+
+In this case we're using the `SLSQP` algorithm, which is a gradient based optimization approach.
+Up to this point none of our components have provided any analytic derivatives,
+so we'll just finite-difference across the whole model to approximate the derivatives.
+That is accompished by this line of code:
+
+.. code::
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/basic_guide/first_mdao.rst]: This code line isn't showing up on ReadTheDocs, but maybe it's not the latest version of your pull up there.

File: openmdao/docs/theory_manual/implicit_transformation_of_vars.rst [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+There are two main component types:
+
+    * :ref:`ExplicitComponent <openmdao.core.explicitcomponent.py>`
+    * :ref:`ImplicitComponent <openmdao.core.implicitcomponent.py>`
+
+The :code:`ExplicitComponent` allows you to define your equations in the explicit form (e.g. :math:1z = \sin(y)`) and it computes the implicit transformation for you in order to compute the residuals.
+The :code:`ImplicitComponent` expects you to compute all the residuals yourself in the :code:`apply_linear` method.
+Regardless of which type of component you chose, OpenMDAO sees everything in the implicit form, and treats your model as system of nonlinear equations.
+
+Multiple components can be aggregated into a hierarchy of with the :code:`Group` class.
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/theory_manual/implicit_transformation_of_vars.rst]: hieararchy of ___ (misssing word)

File: openmdao/docs/theory_manual/implicit_transformation_of_vars.rst [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+A Group is seen by OpenMDAO as a collection of all the implicit equations from all of its children components.
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/theory_manual/implicit_transformation_of_vars.rst]: nonlienar -> nonlinear

File: openmdao/docs/theory_manual/implicit_transformation_of_vars.rst [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+Since both :code:`Component` and :code:`Group` represent systems of nonlienar equations,
+you call the :ref:`add_system <feature_adding_subsystem_to_a_group>` method to construct a model hierarchy.
+
+.. note::
+
+    The underlying mathematics that power OpenMDAO are based on the MAUD_ architecture, which established the foundation
+    for treating a multidisciplinary model as a single system of nonlinear equations as a means to efficiently computing
+    analytic derivatives across a large multidisciplinary model.
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/theory_manual/implicit_transformation_of_vars.rst]: Probably don't need to say "large multidiscplinary model" twice in the same sentence. Reword with "across it".

File: openmdao/test_suite/test_examples/basic_opt_paraboloid.py [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+
+        # build the model
+        prob = Problem()
+        indeps = prob.model.add_subsystem('indeps', IndepVarComp())
+        indeps.add_output('x', 3.0)
+        indeps.add_output('y', -4.0)
+
+        prob.model.add_subsystem('parab', Paraboloid())
+
+        # New line define the component whos output will be constrained
```
Comment (swryan) [inline openmdao/test_suite/test_examples/basic_opt_paraboloid.py]: `# New line   ??`

File: openmdao/test_suite/test_examples/basic_opt_paraboloid.py [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+
+        # setup the optimization
+        prob.driver = ScipyOptimizer()
+        prob.driver.options['optimizer'] = 'COBYLA'
+
+        prob.model.add_design_var('indeps.x', lower=-50, upper=50)
+        prob.model.add_design_var('indeps.y', lower=-50, upper=50)
+        prob.model.add_objective('parab.f_xy')
+
+        # New line to add the constraint to the model
```
Comment (swryan) [inline openmdao/test_suite/test_examples/basic_opt_paraboloid.py]: `# New line   ??`

File: openmdao/docs/basic_guide/first_optimization.rst [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+.. code::
+
+    prob.driver = ScipyOptimizer()
+    prob.driver.options['optimizer'] = 'COBYLA'
+
+Defining the Design Variables and Objective
+---------------------------------------------------------------
+
+Next we setup the problem formulation so the optimizer knows what to vary and what objective to optimize.
```
Comment (swryan) [inline openmdao/docs/basic_guide/first_optimization.rst]: "set up" (setup is a noun)

File: openmdao/docs/basic_guide/first_mdao.rst (hunk 9 above)
Comment (kmarsteller) [reply openmdao/docs/basic_guide/first_mdao.rst]: RTD isn't going to see that until his pull is accepted, right?

Comment (Kenneth-T-Moore) [reply openmdao/docs/basic_guide/first_mdao.rst]: Sorry, I meant BlueDocs. Justin pushed a version of this up there.

File: openmdao/docs/getting_started/index.rst [hunk 16]
```diff
@@ -18,12 +16,17 @@ Use :code:`pip` to install openmdao locally:
 
 :code:`pip install .`
 
-Documentation Building Instructions:
+Hello world!
+*******************
+Here is a really short run file to get you started running your first optimization.
+Copy the code into a file named `hello_world.py` and run it by typing:
+
```
[the remark this answers is not in the document]
Comment (JustinSGray) [reply openmdao/docs/getting_started/index.rst]: yeah. There is a story for allowing us to control how the embed directive handles this. We'll change once the embed directive is updated

File: openmdao/docs/basic_guide/sellar.rst [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+.. note::
+    OpenMDAO comes with other nonlinear solvers you can use if they suit your problem.
+    See the full list :ref:`here <feature_nonlinear_solvers>`
+
+
+The sub-group, named :code:`cycle`, is useful here, because it contains the multi-disciplinary coupling of the Sellar problem.
+This allows us to assign the non-linear solver to :code:`cycle` to just converge those two components, before moving on to the final
+calculations for the :code:`obj_cmp`, :code:`con_cmp1`, and :code:`con_cmp2` to compute the actual outputs of the problem.
+
+Promoting variables with the same name connects them
```
[the remark this answers is not in the document]
Comment (JustinSGray) [reply openmdao/docs/basic_guide/sellar.rst]: these are sub-sub titles. I didn't use title case for them. They are more like emphasis

File: openmdao/docs/basic_guide/sellar.rst [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+-----------------------------------------------------
+
+The data connections in this model are made via promotion.
+OpenMDAO will look at each level of the hierarchy and find all the connect all output-input pairs with the same name.
+
+
+ExecComp is a helper component for quickly defining components for simple equations
```
[the remark this answers is not in the document]
Comment (JustinSGray) [reply openmdao/docs/basic_guide/sellar.rst]: yeah, that was my thought
